hack/README.md

```diff
@@ -1,3 +1,150 @@
 # Hack / maintenance scripts for internal use only
 
 This directory is intended for maintainers/developers
+
+## bash library: `hack/base.bash`
+
+### Features by example
+
+#### Assumptions
+
+1. The script is going to be under `hack/`
+
+#### Integrating
+
+**i. load the lib somewhere at the top of the script:**
+
+```bash
+# --- loading the library ---
+SCRIPT_DEBUG="${SCRIPT_DEBUG:-"0"}"
+TOPDIR="${TOPDIR:-"$(git rev-parse --show-toplevel || true)"}"
+source "${TOPDIR}/hack/base.bash"
+# --- end loading the library ---
+```
+
+Now you can call its functions
+
+**ii. initialize good `FIND`, `GREP`, `SED` values:**
+
+```bash
+utils.tools_setup
+```
+
+### 0. Usage
+
+### 1. logging
+
+#### Default behavior
+
+```bash
+host:ocp user$ source ./hack/base.bash
+host:ocp user$ echo "LOG_LEVEL: '${LOG_LEVEL}'"
+LOG_LEVEL: 'INFO'
+host:ocp user$ log.info "The info message"   # stdout [[ "${LEVELS_MAP[INFO]}" -ge "${LEVELS_MAP["${LOG_LEVEL}"]}" ]]
+Thu Jul 24 14:07:01 UTC 2025 - INFO - The info message
+host:ocp user$ log.debug "The debug message" # stderr [[ "${LEVELS_MAP[DEBUG]}" -ge "${LEVELS_MAP["${LOG_LEVEL}"]}" ]]
+host:ocp user$ log.warning "The my warning"  # stderr [[ "${LEVELS_MAP[WARNING]}" -ge "${LEVELS_MAP["${LOG_LEVEL}"]}" ]]
+Thu Jul 24 14:07:01 UTC 2025 - WARNING - The my warning
+host:ocp user$ log.fatal "The fatality of life" # stderr
+Thu Jul 24 14:07:01 UTC 2025 - FATAL - The fatality of life
+```
+
+#### Controlling the log level
+
+| Option # | Preference | Function        | Variable       | Control           | Comments                               |
+| -------- | ---------- | --------------- | -------------- | ----------------- | -------------------------------------- |
+| `i.`     | Default    | N/A             | `SCRIPT_DEBUG` | +Good             | `0: INFO`, `1: DEBUG`,`2: TRACE`       |
+| `ii.`    | Proper     | `log.set_level` | `LOG_LEVEL`    | +Full,+Safeguards | [`LEVELS_MAP``](hack/base.bash#L15)    |
```

Comment:
In the branch:
https://github.com/mvk/ansible-collection-redhatci-ocp/blob/mac-compat-hack-scripts-lib/hack/base.bash#L486